etl/regenstrief_s3_sftp.py

```diff
@@ -78,68 +67,107 @@ def lambda_handler(event, context):
 
     with transport:
         for s3_file in s3_files(event):
-            filename = s3_file.key
-            bucket = s3_file.bucket_name
-            contents = ""
+
+            # Get the current date in YYYYMMDD format
+            current_date = strftime("%Y_%m_%d")
+            filename = f"midrc_input/{current_date}/{s3_file.key}"
             try:
-                logger.info(f"S3-SFTP: Transferring S3 file '{s3_file.key}'")
+                logger.info(f"S3-SFTP: Transferring S3 file '{filename}' to SFTP")
                 transfer_file(sftp_client, s3_file, filename)
-            except botocore.exceptions.BotoCoreError as ex:
+            except BotoCoreError as ex:
                 logger.exception(
```

Comment:
shouldn't we also `raise` so the lambda execution actually fails, instead of just logging?

Reply:
If we’re uploading multiple files, it might not be ideal to stop the entire process just because one file throws an exception.
What I can do instead is keep track of all the records that fail during the upload and raise a summarized error at the end, highlighting which ones failed.

Reply:
That's fine too